network/wsNetwork_test.go

```diff
@@ -1072,6 +1072,335 @@ func TestGetPeers(t *testing.T) {
 	assert.Equal(t, expectAddrs, peerAddrs)
 }
 
+// mock an identityTracker
+type dummyIdentityTracker struct {
+	shouldInsert bool
+	setCount     int
+	insertCount  int
+	removeCount  int
+	lock         deadlock.Mutex
+}
+
+func newDummyIdentTracker() *dummyIdentityTracker {
+	return &dummyIdentityTracker{
+		shouldInsert: true,
+		setCount:     0,
+		insertCount:  0,
+		removeCount:  0,
+	}
+}
+
+func (d *dummyIdentityTracker) setShouldInsert(b bool) {
+	d.lock.Lock()
+	defer d.lock.Unlock()
+	d.shouldInsert = b
+}
+func (d *dummyIdentityTracker) removeIdentity(p *wsPeer) {
+	d.lock.Lock()
+	defer d.lock.Unlock()
+	d.removeCount++
+}
+func (d *dummyIdentityTracker) getInsertCount() int {
+	d.lock.Lock()
+	defer d.lock.Unlock()
+	return d.insertCount
+}
+func (d *dummyIdentityTracker) getRemoveCount() int {
+	d.lock.Lock()
+	defer d.lock.Unlock()
+	return d.removeCount
+}
+func (d *dummyIdentityTracker) getSetCount() int {
+	d.lock.Lock()
+	defer d.lock.Unlock()
+	return d.setCount
+}
+func (d *dummyIdentityTracker) setIdentity(p *wsPeer) bool {
+	d.lock.Lock()
+	defer d.lock.Unlock()
+	d.setCount++
+	if d.shouldInsert {
+		d.insertCount++
+	}
+	return d.shouldInsert
+}
+
+// TestPeeringWithIdentityChallenge tests the happy path of connecting with identity challenge:
+// - both peers have correctly set ConnectionDeduplicationName
+// - both should exchange identities and verify
+// - both peers should be able to deduplicate connections
+func TestPeeringWithIdentityChallenge(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	netA := makeTestWebsocketNode(t)
+	netA.identTracker = newDummyIdentTracker()
+	addrA := "127.0.0.1:8901"
```

Comment:
Other tests here don't set specific ports, but go with whatever random port they get. I think a fixed port is more brittle.